src/components/buttons/IconButton/index.tsx

```diff
@@ -72,51 +79,70 @@ const ButtonIcon = ({ icon: Icon, color, onClick, disabled, fontSize }) => {
 
     return (
         <Icon
-            {...iconProps}
             color={color === 'white' ? 'inherit' : color}
             style={iconStyles}
             fontSize={fontSize}
         />
     );
 };
-ButtonIcon.defaultProps = {
-    onClick: null,
-    disabled: false,
-    fontSize: 'medium',
-};
-ButtonIcon.propTypes = {
-    onClick: PropTypes.func,
-    icon: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
-    color: PropTypes.string.isRequired,
-    disabled: PropTypes.bool,
-    fontSize: PropTypes.oneOf([
-        'small',
-        'medium',
-        'large',
-        'default',
-        'inherit',
-    ]),
+
+export type IconButtonBuiltInIcon =
+    | 'delete'
+    | 'filter-list'
+    | 'call-merge'
+    | 'remove-red-eye'
+    | 'restore-from-trash'
+    | 'edit'
+    | 'history'
+    | 'map'
+    | 'xml'
+    | 'dhis'
+    | 'orgUnit'
+    | 'refresh'
+    | 'stop'
+    | 'xls'
+    | 'download'
+    | 'globe'
+    | 'clear'
+    | 'clearAll';
+
+type Props = {
+    size?: 'small' | 'medium' | 'large';
+    onClick?: () => void;
+    url?: string;
+    disabled?: boolean;
+    icon?: IconButtonBuiltInIcon;
+    color?: string;
+    overrideIcon?: SvgIconComponent;
+    tooltipMessage: IntlMessage;
+    id?: string;
+    dataTestId?: string;
+    iconSize?: 'small' | 'medium' | 'large' | 'default' | 'inherit';
```

Comment:
FontSize and iconSize are very similar, we can maybe provide a specific type for this and use it in Iaso, Mui is only using string